theatre/studio/src/uiComponents/simpleContextMenu/useCopyPasteKeyframesItem.ts

```diff
@@ -0,0 +1,110 @@
+import {useMemo} from 'react'
+import {useVal} from '@theatre/react'
+import getStudio from '@theatre/studio/getStudio'
+import {useFrameStampPositionD} from '@theatre/studio/panels/SequenceEditorPanel/FrameStampPositionProvider'
+import type {
+  SequenceEditorTree_PrimitiveProp,
+  SequenceEditorTree_PropWithChildren,
+  SequenceEditorTree_SheetObject,
+} from '@theatre/studio/panels/SequenceEditorPanel/layout/tree'
+import {getCopiedKeyframes, getTracks} from '@theatre/studio/selectors'
+import type {IContextMenuItem} from './useContextMenu'
+import type {Keyframe} from '@theatre/core/projects/store/types/SheetState_Historic'
+import type {DopeSheetSelection} from '@theatre/studio/panels/SequenceEditorPanel/layout/layout'
+
+type LeafType =
+  | SequenceEditorTree_SheetObject
+  | SequenceEditorTree_PropWithChildren
+  | SequenceEditorTree_PrimitiveProp
+
+export const usePasteKeyframesItem = (
+  leaf: LeafType,
+): IContextMenuItem | null => {
+  const [posInUnitSpace] = useVal(useFrameStampPositionD())
+  const copiedKeyframes = getCopiedKeyframes()
+  const totalKeyframes = useMemo(() => {
+    return Object.values(copiedKeyframes).reduce((prev, curr) => {
+      return prev + curr.length
+    }, 0)
+  }, [copiedKeyframes])
+
+  return useMemo(() => {
+    const {trackId, sheetObject} = leaf
+
+    if (!trackId || !totalKeyframes) return null
+
+    return {
+      label: `Paste ${totalKeyframes} keyframe${totalKeyframes > 1 ? 's' : ''}`,
+      callback: () => {
+        getStudio().pasteKeyframes({
+          trackId,
+          sheetObject,
+          keyframes: copiedKeyframes,
+          posInUnitSpace,
+        })
+      },
+    }
+  }, [leaf, posInUnitSpace, copiedKeyframes, totalKeyframes])
+}
+
+export const useCopyKeyframesItem = ({
```

Comment:
Not sure why this is a hook. It's (1) not using react APIs, and the memo it generates wouldn't be used untilt he user r-clicks on a keyframe or connector, right?